labelbox/schema/bulk_import_request.py

```diff
@@ -113,6 +115,79 @@ class BulkImportRequest(DbObject):
     project = Relationship.ToOne("Project")
     created_by = Relationship.ToOne("User", False, "created_by")
 
+    @property
+    def inputs(self) -> List[Dict[str, Any]]:
+        """
+        Inputs for each individual annotation uploaded.
+        This should match the ndjson annotations that you have uploaded. 
+        
+        Returns:
+            Uploaded ndjson.
+
+        * This information will expire after 24 hours.    
+        """
+        return self._fetch_remote_ndjson(self.input_file_url)
+
+    @property
+    def errors(self) -> List[Dict[str, Any]]:
+        """
+        Errors for each individual annotation uploaded. This is a subset of statuses
+
+        Returns:
+            List of dicts containing error messages. Empty list means there were no errors
+            See `BulkImportRequest.statuses` for more details.
+
+        * This information will expire after 24 hours.        
+        """
+        self.wait_until_done()
+        return self._fetch_remote_ndjson(self.error_file_url)
+
+    @property
+    def statuses(self) -> List[Dict[str, Any]]:
+        """
+        Status for each individual annotation uploaded.
+
+        Returns:
+            A status for each annotation if the upload is done running.
+            See below table for more details
+            
+        .. list-table::
+           :widths: 15 150
+           :header-rows: 1 
+
+           * - Field
+             - Description
+           * - uuid 
+             - Specifies the annotation for the status row.
+           * - dataRow
+             - JSON object containing the Labelbox data row ID for the annotation.
+           * - status
+             - Indicates SUCCESS or FAILURE.
+           * - errors
+             - An array of error messages included when status is FAILURE. Each error has a name, message and optional (key might not exist) additional_info.
+
+        * This information will expire after 24 hours.        
+        """
+        self.wait_until_done()
+        return self._fetch_remote_ndjson(self.status_file_url)
+
+    @functools.lru_cache()
+    def _fetch_remote_ndjson(self, url: str) -> List[Dict[str, Any]]:
+        """
+        Fetches the remote ndjson file and caches the results.
+
+        Args:
+            url (str): Can be any url pointing to an ndjson file.
+        Returns:
+            ndjson as a list of dicts.
+        """
+        if url is None:
```

Comment:
This null check is a little funny to me, in that the type parameter is only `str` and not `Optional[str]`

if you are going to do this here, should you null check every single paramter passed to every singel function?

Reply:
yeah makes sense. I'll remove it.